### PR TITLE
Add x_state to ui params

### PR DESCRIPTION
**Pull request by tung2744**

### CODE REVIEW

File: pkg/auth/handler/api/workflow_new.go [hunk 1]
```diff
@@ -145,6 +146,7 @@ func (h *WorkflowNewHandler) makeSessionOptions(cookie *http.Cookie) (*workflow.
 		RedirectURI:              uiInfo.RedirectURI,
 		SuppressIDPSessionCookie: uiInfo.SuppressIDPSessionCookie,
 		State:                    uiInfo.State,
+		XState:                   req.XState(),
```
Comment (louischan-oursky) [inline pkg/auth/handler/api/workflow_new.go:149]: Add XState to UIInfo and read uiInfo.XState to set this field, just like what we did for State.

**Review by louischan-oursky — COMMENTED** [review]
We have something called WebhookState in interaction. WebhookState is ultimately in the webhook event context. Therefore we should also expose x_state to hooks.

This is a bit tricky because the handling of WebhookState has issues.

- WebhookState only works for some interaction intents (via IntentWithWebhookState interface). Eliminate IntentWithWebhookState, and read from UIParam.
- Remove WebhookState from webapp session; read from UIParam.

Can you double check if my proposed approach works?

**Comment by tung2744** [comment]
@louischan-oursky For the webhook state, we should now expose both state and x_state in `OAuthContext`, am I correct?

**Comment by louischan-oursky** [comment]
Yes.

**Comment by louischan-oursky** [comment]
Build failed.

**Pull request merged**